src/combinatorics/simple_graph/ends/defs.lean

```diff
@@ -0,0 +1,256 @@
+/-
+Copyright (c) 2022 Anand Rao, Rémi Bottinelli. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anand Rao, Rémi Bottinelli
+-/
+import combinatorics.simple_graph.basic
+import combinatorics.simple_graph.connectivity
+import data.set_like.basic
+import category_theory.category.basic
+import category_theory.filtered
+import topology.category.Top.limits
+
+/-!
+# Ends
+
+This file contains a definition of the ends of a simple graph, as sections of the inverse system
+assigning, to each finite set of vertices, the connected components of its complement.
+-/
+
+universes u
+variables {V : Type u} (G : simple_graph V) (K L L' M : set V)
+
+namespace simple_graph
+
+section out
+
+/-- The graph induced by removing `K` -/
+@[reducible] def out := G.induce $ Kᶜ
+
+/-- Subsetship induces an obvious map on the induced graphs. -/
+@[reducible] def out_hom {K L} (h : K ⊆ L) : G.out L →g G.out K :=
+{ to_fun := λ v, ⟨v.val, set.compl_subset_compl.mpr h v.prop⟩,
+  map_rel' := by { rintros ⟨_, _⟩ ⟨_, _⟩, exact id } }
+
+lemma out_hom_refl (K) : G.out_hom (subset_refl K) = hom.id :=
+by { ext ⟨_, _⟩, refl, }
+
+lemma out_hom_trans {K L M} (h : K ⊆ L) (h' : L ⊆ M) :
+  G.out_hom (h.trans h') = (G.out_hom h).comp (G.out_hom h') :=
+by { ext ⟨_, _⟩, refl, }
+
+lemma out_hom_injective {K} {L} (h : K ⊆ L) : function.injective (G.out_hom h) :=
+by { rintros ⟨v, _⟩ ⟨w, _⟩ e,
+    simpa only [out_hom, subtype.mk_eq_mk, rel_hom.coe_fn_mk] using e, }
+
+end out
+
+/-- The components outside a given set of vertices `K` -/
+@[reducible] def comp_out := (G.out K).connected_component
```

Comment:
This is not a great name. I read it as "compose out".

Reply:
Does `conn_comp_out` seem better?